rest/src/main/java/org/jboss/pnc/rest/endpoint/BuildExecutionEndpoint.java

```diff
@@ -0,0 +1,148 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jboss.pnc.rest.endpoint;
+
+import com.wordnik.swagger.annotations.Api;
+import com.wordnik.swagger.annotations.ApiParam;
+import org.jboss.pnc.auth.AuthenticationProvider;
+import org.jboss.pnc.core.builder.executor.BuildExecutionTask;
+import org.jboss.pnc.core.builder.executor.BuildExecutor;
+import org.jboss.pnc.model.BuildConfiguration;
+import org.jboss.pnc.model.BuildConfigurationAudited;
+import org.jboss.pnc.model.IdRev;
+import org.jboss.pnc.model.User;
+import org.jboss.pnc.rest.provider.BuildRecordProvider;
+import org.jboss.pnc.rest.utils.BpmCallback;
+import org.jboss.pnc.spi.BuildStatus;
+import org.jboss.pnc.spi.datastore.Datastore;
+import org.jboss.pnc.spi.datastore.repositories.BuildConfigurationAuditedRepository;
+import org.jboss.pnc.spi.datastore.repositories.BuildConfigurationRepository;
+import org.jboss.pnc.spi.exception.BuildConflictException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.inject.Inject;
+import javax.servlet.http.HttpServletRequest;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.UriBuilder;
+import javax.ws.rs.core.UriInfo;
+import java.net.URI;
+import java.util.Arrays;
+import java.util.Set;
+import java.util.function.Consumer;
+import java.util.stream.Collectors;
+
+@Api(value = "/builder", description = "Build tasks.")
+@Path("/builder")
+@Produces(MediaType.APPLICATION_JSON)
+@Consumes(MediaType.APPLICATION_JSON)
+public class BuildExecutionEndpoint {
+
+    private static final Logger logger = LoggerFactory.getLogger(BuildExecutionEndpoint.class);
+    private BuildExecutor buildExecutor;
+
+    @Context
+    private HttpServletRequest httpServletRequest;
+
+    private Datastore datastore;
+    private BuildConfigurationRepository buildConfigurationRepository;
+    private BuildConfigurationAuditedRepository buildConfigurationAuditedRepository;
+    private BpmCallback bpmCallback;
+    private BuildRecordProvider buildRecordProvider;
+
+    @Inject
+    public BuildExecutionEndpoint(BuildExecutor buildExecutor, Datastore datastore, BuildConfigurationRepository buildConfigurationRepository, BpmCallback bpmCallback, BuildRecordProvider buildRecordProvider) {
+        this.buildExecutor = buildExecutor;
+        this.datastore = datastore;
+        this.buildConfigurationRepository = buildConfigurationRepository;
+        this.bpmCallback = bpmCallback;
+        this.buildRecordProvider = buildRecordProvider;
+    }
+
+    @Deprecated
+    public BuildExecutionEndpoint() {} // CDI workaround
+
+    @POST
+    @Path("/{taskId}/build")
+    @Consumes(MediaType.WILDCARD)
```

Comment:
Ok my bad - I've just realized that after refactoring the rest of the endpoints...

Since this is an empty POST, adding headers might be tricky. Let's leave it as is.